ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -53,1019 +57,618 @@ namespace ng = ngraph;
-                   std::vector<TensorShape>& input_shapes,
-                   std::vector<const Tensor*>& static_input_map,
-                   ng::runtime::Backend*& op_backend) {
-    std::stringstream signature_ss;
-    string signature;
+  NGRAPH_VLOG(2) << "Deleting from Output Copy Index map " << name();
+  NGraphCatalog::DeleteFromEncapOutputCopyIndexesMap(ng_encap_impl.GetGraphId(),
+                                                     name());
+
+  // Remove entries related to inputs
+  for (int i = 0; i < ng_encap_impl.GetOutputs(); i++) {
+    string key =
+        NGraphCatalog::CreateNodeKey(ng_encap_impl.GetGraphId(), name(), i);
+    if (NGraphCatalog::ExistsInInputVariableSharedNameMap(key)) {
+      NGraphCatalog::DeleteFromInputVariableSharedNameMap(key);
+      NGRAPH_VLOG(2) << "Deleting from input variable shared name map " << key;
+    }
+  }
 
-    std::shared_ptr<ngraph::Function> ng_function;
-    std::shared_ptr<ngraph::runtime::Executable> evicted_ng_exec;
+#endif
 
-    NGRAPH_VLOG(4) << "GetNgExec: Got backend of type: " << m_op_backend_name;
-    op_backend = BackendManager::GetBackend(m_op_backend_name);
+  ng_encap_impl.ClearNgExecInputCache();
+  ng_encap_impl.ClearNgExecOutputCache();
+  ng_encap_impl.ClearNgExecMap();
+  ng_encap_impl.ClearNgFunctionMap();
+
+  // Release the backend
+  NGRAPH_VLOG(2) << "~NGraphEncapsulateOp():: ReleaseBackend";
+  BackendManager::ReleaseBackend(ng_encap_impl.GetOpBackend());
+  event.Stop();
+  ngraph::Event::write_trace(event);
+}
+
+//---------------------------------------------------------------------------
+// OpKernel::Compute
+//---------------------------------------------------------------------------
+void NGraphEncapsulateOp::Compute(OpKernelContext* ctx) {
+  std::ostringstream oss;
+  oss << "Execute: Encapsulate_" << ng_encap_impl.GetInstanceId() << ": "
+      << name();
+  ngraph::Event event(oss.str(), name(), "");
+
+  Timer compute_time;
+  std::lock_guard<std::mutex> lock(m_compute_lock);
+  NGRAPH_VLOG(4) << "NGraphEncapsulateOp::Compute starting for cluster "
+                 << ng_encap_impl.GetNgraphCluster();
+
+  ngraph::Event event_func_maybe_create("FunctionMaybeCreate", name(), "");
+  Timer function_lookup_or_create;
+
+  std::vector<TensorShape> input_shapes;
+  std::vector<const Tensor*> static_input_map;
+  std::shared_ptr<ngraph::Function> ng_function;
+  std::shared_ptr<ngraph::runtime::Executable> ng_exec;
+  ng::runtime::Backend* op_backend;
+
+  // TF input tensor
+  std::vector<Tensor> tf_input_tensors;
+
+  for (int i = 0; i < ctx->num_inputs(); i++) {
+    tf_input_tensors.push_back(ctx->input(i));
+  }
 
-    // Compute Signature
-    TF_RETURN_IF_ERROR(
-        ComputeSignature(ctx, signature_ss, input_shapes, static_input_map));
-    signature = signature_ss.str();
+  int step_id = ctx->step_id();
+
+  // Get ngraph executable and inputs information
+  OP_REQUIRES_OK(ctx, ng_encap_impl.GetNgExecutable(
```

Comment:
GetNgExecutable() does not need to take in this inputs. They are not used anywhere else in the Compute function call.
std::vector<TensorShape> input_shapes;
std::vector<const Tensor*> static_input_map;
Can be made a TODO to be addressed in another PR

Reply:
I am calling ComputeSignature inside GetNgExecutable, so need these inputs for that. 
We discussed this before, I ran into issue when i tried to make these separate, maybe we can address in future PR

Reply:
Yes, i remember this discussion. :) We can add a jira ticket to look into this. This will address @sayantan-nervana s comment also on ComputeSignature.